src/java/arjdbc/postgresql/PGDriverImplementation.java

```diff
@@ -164,6 +170,32 @@ private static ByteList formatInterval(final Object object) {
         return str;
     }
 
+    private static Class<?> internalPGArrayClass;
+    private static Class<?> internalPGArrayClass(final ThreadContext context) {
+        final Class<?> clazz = internalPGArrayClass;
+        if (clazz == null) {
+            final Ruby runtime = context.getRuntime();
+            try {
+                // load older (internal) Jdbc4Array Class (9.2 and earlier)
+                internalPGArrayClass = runtime.getJavaSupport().loadJavaClass("org.postgresql.jdbc4.Jdbc4Array");
+                return internalPGArrayClass;
+            }
+            catch (ClassNotFoundException e) { /* should never be reached */ }
+        }
+        return clazz;
+    }
+
+    private static boolean shouldUseInternalPGArrayClass() {
+        try {
+            // try loading official PgArray Class (since postgresql 9.4)
+            Class.forName("import org.postgresql.jdbc.PgArray");
```

Comment:
This is heavily WiP and was indeed not tested at all - i merely committed because i had to leave.
This was never intended to get actually pushed to GitHub...

Sorry for wasting your time 😔